editor.planx.uk/src/routes/views/publishedPreview.tsx

```diff
@@ -0,0 +1,63 @@
+import { FlowGraph } from "@opensystemslab/planx-core/types";
+import axios, { AxiosError } from "axios";
+import { NaviRequest, NotFoundError } from "navi";
+import { useStore } from "pages/FlowEditor/lib/store";
+import PublicLayout from "pages/layout/PublicLayout";
+import SaveAndReturnLayout from "pages/layout/SaveAndReturnLayout";
+import React from "react";
+import { View } from "react-navi";
+import { getTeamFromDomain, setPath } from "routes/utils";
+
+import { fetchDataForPublishedView } from "./published";
+
+/**
+ * View wrapper for /publish-preview
+ * Fetches all necessary data, and sets up Save & Return layout
+ */
+export const publishedPreviewView = async (req: NaviRequest) => {
+  const flowSlug = req.params.flow.split(",")[0];
+  const teamSlug =
+    req.params.team || (await getTeamFromDomain(window.location.hostname));
+
+  // /publish-preview uses the same theme & global settings as /preview
+  const data = await fetchDataForPublishedView(flowSlug, teamSlug);
+  const flow = data.flows[0];
+  if (!flow)
+    throw new NotFoundError(`Flow ${flowSlug} not found for ${teamSlug}`);
+
+  // /publish-preview fetches draft data of this flow and the latest published version of each external portal
+  const flowData = await fetchFlattenedFlowData(flow.id);
+
+  const state = useStore.getState();
+  state.setFlow({ id: flow.id, flow: flowData, flowSlug });
+  state.setFlowNameFromSlug(flowSlug);
+  state.setGlobalSettings(data.globalSettings[0]);
+  state.setFlowSettings(flow.settings);
+  state.setTeam(flow.team);
+
+  return (
+    <PublicLayout>
+      <SaveAndReturnLayout>
+        <View />
+      </SaveAndReturnLayout>
```

Comment:
I'm choosing to use the `<SaveAndReturnLayout>` for `/publish-preview` which means a lowcal session id will be generated and saved. 

`/unpublished` does _not_ currently use this layout which explains a number of the "cannot find session id" errors on production lately around CSV generation - do we want to make a decision to use _everywhere_?

Reply:
Thanks for getting to the bottom of this - for consistency I'd say we use it everywhere. I think this means we can hoist it higher up the component tree. Maybe it can just be merged into `PublicLayout`?

Reply:
Slept on this one and having a few doubts that we should use `<SaveAndReturnLayout>` throughout rather than just clean up/improve our error handling for integrations that fail without a `sessionId` or `saveToEmail` on `/unpublished` & `/publish-preview` routes ? 

A couple considerations / possible side effects:
- Would this unnecessarily bloat `lowcal_sessions` records, making it more confusing to summarise analytics etc? Would we need to add an additional column to filter by editor route to get just the "real" `/preview` sessions? 
- Neither `/unpublished` or `/public-preview` load with `analytics=false` but will now have a valid `sessionId` in store and possibly create records? I didn't follow this thread too carefully through code, but feels a bit risky? 
- With the email login screen on non-all-published routes, we basically open ourselves up to the risk again that this change in publishing logic is meant to protect against - which is an application being submitted against work-in-progress content that wasn't intended for public use yet? Feels easier for an editor to confusingly share wrong link, etc?

I think I'd like to exclusively use `<SaveAndReturnLayout>` on `/preview` and revert original decision to also wrap `/publish-preview` here. Will pick up error handling in a separate followup PR if this sounds good?

Reply:
Thanks for thinking on this one - this seems sensible to me. I think the fact that we've narrowed it down to this is actually already super helpful when it comes to debugging etc, so let's keep as is for now 👍